src/WebJobs.Script.WebHost/Extensions/ZipArchiveExtensions.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Abstractions;
+using System.IO.Compression;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Extensions
+{
+    public static class ZipArchiveExtensions
+    {
+        public static async Task AddDirectory(this ZipArchive zipArchive, DirectoryInfoBase directory, string directoryNameInArchive)
+        {
+            await InternalAddDirectory(zipArchive, directory, directoryNameInArchive);
+        }
+
+        private static async Task InternalAddDirectory(ZipArchive zipArchive, DirectoryInfoBase directory, string directoryNameInArchive, IList<ZipArchiveEntry> files = null)
+        {
+            bool any = false;
+            foreach (var info in directory.GetFileSystemInfos())
+            {
+                any = true;
+                if (info is DirectoryInfoBase subDirectoryInfo)
+                {
+                    string childName = ForwardSlashCombine(directoryNameInArchive, subDirectoryInfo.Name);
+                    await InternalAddDirectory(zipArchive, subDirectoryInfo, childName, files);
+                }
+                else
+                {
+                    var entry = await zipArchive.AddFile((FileInfoBase)info, directoryNameInArchive);
+                    files?.Add(entry);
+                }
+            }
+
+            if (!any)
+            {
+                // If the directory did not have any files or folders, add a entry for it
+                zipArchive.CreateEntry(EnsureTrailingSlash(directoryNameInArchive));
+            }
+        }
+
+        private static string ForwardSlashCombine(string part1, string part2)
```

Comment:
Is this required? Looks like all paths are passed in from file system lookups and you should have the correct directory separator, right?

Reply:
this is for paths used in the zip file central archive, regardless of the platform, they should always be `/`. I think the Windows Explorer zip handles both separators, same with 7zip, but native unzip on mac and linux don't. Since a zip file can be unzipped on non-windows machines as well, we should always keep it `/`